cmd/kubeadm/app/phases/addons/addons.go

```diff
@@ -45,8 +45,12 @@ func CreateEssentialAddons(cfg *kubeadmapi.MasterConfiguration, client clientset
 		return fmt.Errorf("error when parsing kube-proxy configmap template: %v", err)
 	}
 
+	repoPrefix := cfg.ImageRepository
+	if cfg.CIImageRepository != "" {
```

Comment:
ok. regarding the rest, right now PR is in WIP stage, to see that all API tests are passing. I'll update it with all other comments addressed.